src/algebra/group_power/basic.lean

```diff
@@ -467,6 +467,25 @@ end
 
 end cancel_add_monoid
 
+namespace comm_semiring
+
+variables [comm_semiring R]
+
+lemma min_pow_dvd_sum {n m : ℕ} {a b c : R} (ha : c^n ∣ a) (hb : c^m ∣ b) :
+c^(min n m) ∣ a + b :=
+begin
+by_cases n ≤ m,
+{ simp [h],
+  replace hb := dvd.trans (pow_dvd_pow c h) hb,
+  exact dvd_add ha hb },
+{ simp [le_of_not_ge h],
+  replace ha := dvd.trans (pow_dvd_pow c (le_of_not_ge h)) ha,
+  exact dvd_add ha hb,
+ },
+end
```

Comment:
```suggestion
lemma min_pow_dvd_sum {n m : ℕ} {a b c : R} (ha : c ^ n ∣ a) (hb : c ^ m ∣ b) :
  c ^ (min n m) ∣ a + b :=
begin
  by_cases h : n ≤ m,
  { simp [h],
    replace hb := dvd.trans (pow_dvd_pow c h) hb,
    exact dvd_add ha hb },
  { simp [le_of_not_ge h],
    replace ha := dvd.trans (pow_dvd_pow c (le_of_not_ge h)) ha,
    exact dvd_add ha hb }
end
```

Reply:
I've reformatted the proof according to the style guide.
Other point: I've added `h :` in the `by_cases` line, because you were using an unnamed argument in the remainder of the proof. Typically this isn't a problem, but it's a practice that doesn't scale to a large library.
Similarly, `simp` in the middle of a proof is completely fine on a scratch-pad, but it doesn't scale, for two reasons:
1) If somebody adds a new simp-lemma in one of the files that you are importing, then suddenly this proof might break because the `simp` does something different.
2) It might slow down the library.
The solution is to replace `simp` with the output of `squeeze_simp`.

Note that you can also inline the `replace` lines into the `exact`.

Reply:
An alternative proof without `cases` (untested):
```
begin
  replace ha := (pow_dvd_pow c (min_le_left n m)).trans ha,
  replace hb := (pow_dvd_pow c (min_le_right n m)).trans hb,
  exact dvd_add ha hb
end
```
or a term mode version:
```
dvd_add ((pow_dvd_pow c (min_le_left n m)).trans ha)
  ((pow_dvd_pow c (min_le_right n m)).trans hb)
```

Reply:
```
begin
  replace ha := dvd.trans (pow_dvd_pow c (min_le_left n m)) ha,
  replace hb := dvd.trans (pow_dvd_pow c (min_le_right n m)) hb,
  exact dvd_add ha hb
end
```
Works and it is better than what I wrote.